src/lib/orm/metadata.go

```diff
@@ -30,33 +30,74 @@ var (
 	cache = sync.Map{}
 )
 
-type key struct {
+// Key ...
+type Key struct {
 	Name       string
 	Filterable bool
 	FilterFunc func(context.Context, orm.QuerySeter, string, interface{}) orm.QuerySeter
 	Sortable   bool
 }
 
-type metadata struct {
-	Keys         map[string]*key
-	DefaultSorts []*q.Sort
+// Column the columns of the model
+type Column struct {
+	Name       string // the column name in db
+	Filterable bool   // True when the column is searchable
+	Sortable   bool   // True when the column is orderable
 }
 
-func (m *metadata) Filterable(key string) (*key, bool) {
+// Metadata metadata of model
+type Metadata struct {
+	Keys map[string]*Key
+
+	columnIndexes map[string]*Column
+	Columns       []*Column // ordered columns
```

Comment:
What's the different of `Keys`, `columnIndexes` and `Columns`?

Reply:
I will remove the Keys in the metadata.

`Columns` keeps the order for columns, the selector will use this to generate a stable SQL statement (means using the same model and filters will generate the same SQL).  `map` doesn't work for this case because iterations of the map in golang are random seq when using `range`.

`columnIndexes` is used to get the Column by the db column name or model FieldName. And the items in the `Columns` are all tracked by the db column name, on model FieldName in it.

While, there FieldName, db column name, filter by method in the Keys is not very clear.

Reply:
Now, `Columns` only be the info parsed from the fields of the model and the filterFuncs be the info parsed from the filter methods and filter fields of the model.